realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1093,27 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Returns a distinct set of objects of a specific class. As a Realm is unordered, it is
+     * undefined which objects are returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return a non-null {@link RealmResults} containing the distinct objects.
+     * @throws IllegalArgumentException if a field name does not exist.
+     * @throws IllegalArgumentException if a field's type is not supported.
+     * @throws UnsupportedOperationException if a field is not indexed.
+     */
+    public RealmResults<E> distinct(String fieldName) {
```

Comment:
That is fine. 2 + 3 should probably be in their own PR though instead of keeping this alive for to long.